### PR TITLE
#4137 enabled and extended covid vaccination info on cases (BE + WEB)

### PR DESCRIPTION
**Pull request by leventegal-she**

### CODE REVIEW

File: sormas-api/src/main/resources/validations.properties [hunk 1]
```diff
@@ -139,4 +139,5 @@ externalJournalPersonValidationPhone = Invalid phone number.
 externalJournalPersonValidationPhoneTaken = The phone number is already in use in the symptom journal.
 externalJournalPersonValidationEmailTaken = The email address is already in use in the symptom journal.
 externalJournalPersonValidationNoEmailOrPhone = Please provide an email address or a phone number.
-campaignFormMetaValidationUnexpectedError = There was an unexpected error while trying to validate this form. Please forward the following error message to a system administrator: %s
\ No newline at end of file
+campaignFormMetaValidationUnexpectedError = There was an unexpected error while trying to validate this form. Please forward the following error message to a system administrator: %s
+vaccineDosesFormat="Vaccine dose should be a number between 1 and 10"
```
Comment (MateStrysewske) [inline sormas-api/src/main/resources/validations.properties:143]: Remove the "" signs

File: sormas-api/src/main/resources/enum.properties [hunk 2]
```diff
@@ -1326,4 +1328,24 @@ Salutation.MRS=Dear Madame
...
+Salutation.OTHER=Other
+
+# Vaccine
+Vaccine.PFIZER_BIONTECH=Pfizer–BioNTech COVID‑19 vaccine
+Vaccine.MODERNA=Moderna COVID-19 Vaccine
+Vaccine.OXFORD_ASTRA_ZENECA=Oxford–AstraZeneca COVID-19 vaccine
+Vaccine.Ad26_COV2_S=Ad26.COV2.S
+Vaccine.NVX_COV_2373=Novavax COVID-19 vaccine
+Vaccine.SANOFI_GSK=Sanofi-GSK
+Vaccine.UNKOWN=Unkown
```
Comment (MateStrysewske) [inline sormas-api/src/main/resources/enum.properties:1340]: => UNKNOWN

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import javax.annotation.Nullable;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+
+public enum Vaccine {
+
+	PFIZER_BIONTECH(VaccineManufacturer.PFIZER_BIONTECH),
+	MODERNA(VaccineManufacturer.MODERNA),
+	OXFORD_ASTRA_ZENECA(VaccineManufacturer.ASTRA_ZENECA),
+	Ad26_COV2_S(VaccineManufacturer.JOHNSON_JOHNSON),
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java:27]: put "d" in uppercase as well (not in the translation, just in the enum name)

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+package de.symeda.sormas.api.caze;
+
+import javax.annotation.Nullable;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+
+public enum Vaccine {
+
+	PFIZER_BIONTECH(VaccineManufacturer.PFIZER_BIONTECH),
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java:24]: Rename enum value to `COMIRNATY`

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	MODERNA(VaccineManufacturer.MODERNA),
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/Vaccine.java:25]: Rename enum value to `MRNA_1273`

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/VaccineManufacturer.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import de.symeda.sormas.api.i18n.I18nProperties;
+
+public enum VaccineManufacturer {
+
+	PFIZER_BIONTECH,
+	MODERNA,
+	ASTRA_ZENECA,
+	JOHNSON_JOHNSON,
+	NOVAVAX,
+	SANOFI_GLAXOSMITHKLINE,
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/VaccineManufacturer.java:27]: Rename enum value to SANOFI_GSK

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/VaccineManufacturer.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.api.caze;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+
+public enum VaccineManufacturer {
+
+	PFIZER_BIONTECH,
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/VaccineManufacturer.java:22]: Rename enum value to BIONTECH_PFIZER

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

File: sormas-app/app/src/main/res/layout/fragment_case_edit_layout.xml [hunk 8]
```diff
@@ -692,6 +701,79 @@
...
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:orientation="horizontal">
+                    <de.symeda.sormas.app.component.controls.ControlSpinnerField
+                        android:id="@+id/caseData_vaccineName"
+                        app:value="@={data.vaccineName}"
+                        style="@style/ControlFirstOfTwoColumnsStyle" />
+
+                    <de.symeda.sormas.app.component.controls.ControlSpinnerField
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/res/layout/fragment_case_edit_layout.xml:713]: Needs to be a ControlTextEditField

File: sormas-app/app/src/main/res/layout/fragment_case_edit_layout.xml [hunk 9]
```diff
@@ -692,6 +701,79 @@
...
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:orientation="horizontal">
+                    <de.symeda.sormas.app.component.controls.ControlSpinnerField
+                        android:id="@+id/caseData_vaccineManufacturer"
+                        app:value="@={data.vaccineManufacturer}"
+                        style="@style/ControlFirstOfTwoColumnsStyle" />
+
+                    <de.symeda.sormas.app.component.controls.ControlSpinnerField
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/res/layout/fragment_case_edit_layout.xml:730]: Needs to be a ControlTextEditField

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* Captions for "otherVaccineName" and "otherVaccineManufacturer" are missing
* Add the new fields to the detailed and custom case export (that part was missing in the feature description, sorry!)
* The logic that selects the manufacturer based on the selected vaccine name is executed in the mobile app when opening the edit form, which causes problems when e.g. "Other" vaccine name and "Other" manufacturer are selected; in that case, the manufacturer spinner will be cleared, resulting in data loss; the logic may only be executed when the value is changed by the user

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
* Make sure that "other manufacturer" and "other vaccine name" are cleared when "Other" is deselected in their respective parent field

**Comment by leventegal-she** [comment]
> * Make sure that "other manufacturer" and "other vaccine name" are cleared when "Other" is deselected in their respective parent field

```
			FieldHelper.setVisibleWhen(
				getFieldGroup(),
				CaseDataDto.OTHER_VACCINE_NAME,
				CaseDataDto.VACCINE_NAME,
				Collections.singletonList(Vaccine.OTHER),
				true);

			FieldHelper.setVisibleWhen(
				getFieldGroup(),
				CaseDataDto.OTHER_VACCINE_MANUFACTURER,
				CaseDataDto.VACCINE_MANUFACTURER,
				Collections.singletonList(VaccineManufacturer.OTHER),
				true);

```

This is the setup for hiding the `other...` fields, the last parameter set to `true`, this mieans they will be cleared when they get hidden.

**Pull request merged**